### PR TITLE
Feature/#128 file system access interface

### PR DESCRIPTION
**Pull request by panwrona**
Thank you for your contribution to MagicScript Project.
Before submitting this PR, please make sure:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [x] You have added unit tests

### CODE REVIEW

File: src/platform/lumin-runtime/native-file-system.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class NativeFileSystem {
+    
+    writeFile(path, content, encoding) {
+        return Promise.reject(new Error('Write file function is not implemented yet'));
```
Comment (pleszkiewicz) [inline src/platform/lumin-runtime/native-file-system.js:4]: What Lumin API should be used here? Is it available? Can we implement these functions now?

Comment (kpiascik) [reply src/platform/lumin-runtime/native-file-system.js:4]: @creationix can you help here?

Comment (creationix) [reply src/platform/lumin-runtime/native-file-system.js:4]: The raw FS APIs for magicscript lumin are documented here http://api.magicscript.org/modules/_uv_.fs.html

But these are very low-level.  You probably want to re-use some code from polyfills.

For example, here are implementations of the node.js style writeFile(Sync) functions:
- https://github.com/magic-script/magic-script-polyfills/blob/master/src-ts/node/fs.ts#L573
- https://github.com/magic-script/magic-script-polyfills/blob/master/src-ts/node/fs.ts#L607

To use the node functions, simply import "fs" as long as the polyfills plugin for rollup is turned on.

Also, any code from polyfills can be imported directly like:

```js
// Node.js style APIs
import { writeFileSync, writeFile } from "fs";

// simpler internal functions in polyfills
import { writeFileStream } from "magic-script-polyfills/src/fs-promised.js"
```

Comment (creationix) [reply src/platform/lumin-runtime/native-file-system.js:4]: If you'd like, I can write a function for you.  I just need to know the desired interface.  It looks like you want to return a promise, I'm assuming `path` is a string.  Does `content` only take a string, or does it also take `ArrayBuffer` or `ArrayBufferView`s like `Uint8Array`?  What is `encoding` for and is it actually used anywhere or do we always assume utf8 for strings?

Comment (creationix) [reply src/platform/lumin-runtime/native-file-system.js:4]: Is this the interface you're trying to copy? https://github.com/itinance/react-native-fs#writefilefilepath-string-contents-string-encoding-string-promisevoid

```ts
import { open, write, close } from "magic-script-polyfills/src/fs-promised.js"
import { b64ToBin, rawToBin, strToBin } from "magic-script-polyfills/src/bintools.js"

export async function writeFile(filepath: string, contents: string, encoding: "utf8" | "ascii" | "base64" = "utf8"): Promise<void> {
    const fd = await open(filepath, "w", 0o644);
    try {
        const body: Uint8Array = encoding === 'utf8' ? strToBin(contents)
            : encoding === 'ascii' ? rawToBin(contents) 
            : encoding === 'base64' ? b64ToBin(contents)
            : `Unsupported encoding '${encoding}'`;
        await write(fd, body, 0);
    }
    finally {
        await close(fd);
    }
}
```

File: src/platform/lumin-runtime/native-file-system.js [hunk 2]
```diff
@@ -1,19 +1,42 @@
...
+            await close(fd);
+        }
     }
 
-    readFile(path, encoding) {
-        return Promise.reject(new Error('Read file function is not implemented yet'));
+    async readFile(path, encoding) {
+        const fd = await open(path, "r", 0o444);
+        try {
+            let content = new Uint8Array(64 * 1024);
```
Comment (creationix) [inline src/platform/lumin-runtime/native-file-system.js:26]: This needs to not hard-code the file length unless you know all files are exactly 64k long.

**Pull request merged**